### ISSUE TITLE
upstreaming?

### ISSUE THREAD

**Issue by anarcat** [owentl/gitlab-prometheus#1]
hi!

thanks for doing this, i was looking for something exactly like this. i cannot fathom, however, why this is not part of the [builtin gitlab metrics](https://docs.gitlab.com/ee/administration/monitoring/prometheus/gitlab_metrics.html). there's a [issues statistics API endpoint](https://docs.gitlab.com/ee/api/issues_statistics.html) but no prometheus metrics, weird no?

have you looked upstream to see if that was on the roadmap?

thanks!

**Comment by anarcat** [comment]
i filed [this ticket upstream](https://gitlab.com/gitlab-org/gitlab/-/issues/350602) and linked it to this project as well.

**Comment by owentl** [comment]
I didn't search through the issues on GitLab but I did talk to our GitLab account team and they didn't know of anything outside of what Grafana sells in their Enterprise offering.  As we were only interested in this functionality it seemed a bit pricey for our needs.

I am glad that you found this project useful!

**Comment by owentl** [comment]
> i filed [this ticket upstream](https://gitlab.com/gitlab-org/gitlab/-/issues/350602) and linked it to this project as well.

Well the upstream ticket went nowhere with GitLab’s acquisition of Opstrace, which I get from their perspective.

if you run into any issues setting this project up please let me know, happy to help!

**Comment by anarcat** [comment]
On 2022-01-25 16:30:20, Tyler wrote:
>> i filed [this ticket upstream](https://gitlab.com/gitlab-org/gitlab/-/issues/350602) and linked it to this project as well.
>
> Well the upstream ticket went nowhere with GitLab’s acquisition of Opstrace, which I get from their perspective.

I guess I can comment more liberally here: I think that's really
annoying. Prometheus is not just some random monitoring project, it's
the defactor metrics format. You don't *have* to use prometheus, but I
don't see why you wouldn't *expose* your metrics through it.

Or, to put another way, if Opstrace can't scrape Prometheus exporters,
it's broken. :p But maybe I drank too much prometheus coolaid...

a.

-- 
A people that elect corrupt politicians, imposters, thieves & traitors
are not victims... but accomplices.
                        - George Orwell

**Comment by owentl** [comment]
> I guess I can comment more liberally here: I think that's really annoying. Prometheus is not just some random monitoring project, it's the defactor metrics format. You don't *have* to use prometheus, but I don't see why you wouldn't *expose* your metrics through it. Or, to put another way, if Opstrace can't scrape Prometheus exporters, it's broken. :p But maybe I drank too much prometheus coolaid... a.
> […](#)

I agree with you, personally we have spent too much time and effort into all the Prometheus metrics we have throughout the org, so not looking to recreate that effort!  Not to mention personally like a separation of duties for different tasks.
